src/integration/data/vectors/upsertAndUpdate.test.ts

```diff
@@ -155,13 +157,47 @@ describe('Testing retry logic on Upsert operation, as run on a mock, in-memory h
     // Call Upsert operation
     await mockServerlessIndex.upsert(recordsToUpsert);
 
+    // 2 total tries: 1 initial call, 1 retry
+    expect(retrySpy).toHaveBeenCalledTimes(1); // passes
+    expect(delaySpy).toHaveBeenCalledTimes(1); // fails
+    expect(callCount).toBe(2);
+  });
+
+  test('Update operation should retry 1x if server responds 1x with error and 1x with success', async () => {
```

Comment:
Figured duplicating this type of test across `upsert` and `update` would be enough to justify me _not_ duplicating it again for `configureIndex`, but lmk if you disagree!

Reply:
(We should rly centralize this type of thing to avoid duplicating this logic, but I think this is okay for now)

Reply:
It seems useful to validate that the calls themselves trigger retries as we'd expect, is that what you're talking about centralizing?

Reply:
Mmm I'm not 100% sure we're on the same page -- when you say "the calls themselves," are you talking about the different async funcs that we could pass into the `RetryWrapper`?